packages/diff-sequences/perf/index.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// Make sure to run node with --expose-gc option!
+
+// The times are reliable if about 1% relative mean error if you run it:
+
+// * immediately after restart
+// * with 100% battery charge
+// * not connected to network
+
+/* eslint import/no-extraneous-dependencies: "off" */
+
+const Benchmark = require('benchmark');
+
+const diffBaseline = require('diff').diffLines;
+const diffImproved = require('../build/index.js').default;
+
+const testBaseline = (a, b) => {
+  const benchmark = new Benchmark({
+    fn() {
+      diffBaseline(a, b);
+    },
+    name: 'baseline',
+    onCycle() {
+      global.gc(); // after run cycle
+    },
+    onStart() {
+      global.gc(); // when benchmark starts
+    },
+  });
+
+  benchmark.run({async: false});
+
+  return benchmark.stats;
+};
+
+const testImproved = function(a, b) {
+  const benchmark = new Benchmark({
+    fn() {
+      // Split string arguments to make fair comparison with baseline.
+      const aItems = a.split('\n');
+      const bItems = b.split('\n');
+
+      const isCommon = (aIndex, bIndex) => aItems[aIndex] === bItems[bIndex];
+
+      // This callback obviously does less than baseline `diff` package,
+      // but avoiding double work and memory churn is the goal.
+      // For example, `jest-diff` has had to split strings that `diff` joins.
+      const foundSubsequence = () => {};
+
+      diffImproved(aItems.length, bItems.length, isCommon, foundSubsequence);
+    },
+    name: 'improved',
+    onCycle() {
+      global.gc(); // after run cycle
+    },
+    onStart() {
+      global.gc(); // when benchmark starts
+    },
+  });
+
+  benchmark.run({async: false});
+
+  return benchmark.stats;
+};
+
+const writeHeading2 = () => {
+  process.stdout.write(
```

Comment:
`console.log()`? It's implemented as `process.stdout.write(message + '\n')` (plus interpolation) in node core: https://github.com/nodejs/node/blob/29d3d1ea135c5d5692c3fcd9fb9091efa1db5760/lib/internal/console/constructor.js#L213-L228